libcaf_core/caf/json_reader.test.cpp

```diff
@@ -388,6 +424,22 @@ fixture::fixture() {
   // Test cases for proper handling of whitespace.
   add_test_case("[1, \r\n2,\f\t\v3]", ls<int32_t>(1, 2, 3));
   add_test_case("\r\n{\r\n\"a\":\r\n1, \"b\"\r\n:\r\n2}\r\n", my_request(1, 2));
+  // Test cases for proper handling of code point escape sequences.
+  // Single byte utf-8 sequences.
+  add_test_case(
+    R"_("\u0048\u0065\u006c\u006c\u006f\u002c\u0020\u0022\u0057\u006f\u0072\u006c\u0064\u0022\u0021")_",
+    std::string{R"_(Hello, "World"!)_"});
+  // TODO why do we need two backslashes when the first one is already escaped?
+  add_test_case(R"_("\u005c\u005c")_", std::string{R"_(\)_"});
+  // Two byte utf-8 sequences.
+  add_test_case(R"_("\u0107\u010d\u017e\u0161\u0111")_", std::string{"ćčžšđ"});
+  // Three byte utf - 8 sequences.
+  add_test_case(R"_("\u20AC\u2192\u221E")_", std::string{"€→∞"});
+  // Four byte utf - 8 sequences.
```

Comment:
Stick to one spelling. Separating the name with spaces is rather uncommon, so `utf-8` should be preferred, I think.